catalog/controller/extension/payment/razorpay.php

```diff
@@ -842,4 +867,97 @@ public function update()
             return  $this->response->redirect($this->url->link('extension/payment/razorpay/info', 'subscription_id=' . $postData['subscriptionId'], true));
         }
     }
+
+    protected function processSubscriptionAction($data)
+    {
+        $subscriptionId = $data['payload']['subscription']['entity']['id'];
+
+        if (empty($subscriptionId) === false) {
+            $merchant_order_id = $data['payload']['subscription']['entity']['notes']['merchant_order_id'];
+
+            if(isset($merchant_order_id) === true)
+            {
+                switch ($data['event']) {
+                    case 'subscription.paused':
+                        $status = 'paused';
+                        $oc_status = 2;
+                        break;
+
+                    case 'subscription.resumed':
+                        $status = 'active';
+                        $oc_status = 1;
+                        break;
+
+                    case 'subscription.cancelled':
+                        $status = 'cancelled';
+                        $oc_status = 3;
+                        break;
+                }
+
+                $this->load->model('extension/payment/razorpay');
+
+                $this->model_extension_payment_razorpay->updateSubscriptionStatus($subscriptionId, $status, "Webhook" );
+                $this->model_extension_payment_razorpay->updateOCRecurringStatus($merchant_order_id, $oc_status);
+                $this->log->write("Subscription ".$status." webhook event processed for Opencart OrderID (:" . $merchant_order_id . ")");
+
+                return;
+            }
+        }
+    }
+
+    protected function processSubscriptionCharged($data)
+    {
+
+        $paymentId = $data['payload']['payment']['entity']['id'];
+        $subscriptionId = $data['payload']['subscription']['entity']['id'];
+        $merchant_order_id = $data['payload']['subscription']['entity']['notes']['merchant_order_id'];
+        $webhookSource = $data['payload']['subscription']['entity']['source'];
+        $amount = number_format($data['payload']['payment']['entity']['amount'] / 100, 4, ".", "");
+
+        $this->load->model('extension/payment/razorpay');
+
+        // Process only if its from opencart subscription source
+        if ($webhookSource == "opencart-subscription") {
+
+            $subscription = $this->api->subscription->fetch($subscriptionId)->toArray();
+            $rzpSubscription = $this->model_extension_payment_razorpay->getSubscriptionById($subscriptionId);
+
+            if ($subscription['paid_count'] == 1) {
+                if ($rzpSubscription['status'] == 'authenticated' && $rzpSubscription['paid_count'] == 0) {
```

Comment:
Follow any one of this
if(array_in($rzpSubscription['status'] , ['created','authenticated']) && $rzpSubscription['paid_count'] == 0) {


if (($rzpSubscription['status'] == 'authenticated' || $rzpSubscription['status'] == 'created' )&& $rzpSubscription['paid_count'] == 0) {